src/snowflake/snowpark/_internal/compiler/telemetry_constants.py

```diff
@@ -6,10 +6,19 @@
 
 
 class CompilationStageTelemetryField(Enum):
+    # types
     TYPE_LARGE_QUERY_BREAKDOWN_OPTIMIZATION_SKIPPED = (
         "snowpark_large_query_breakdown_optimization_skipped"
     )
+    TYPE_COMPILATION_STAGE_STATISTICS = "snowpark_compilation_stage_statistics"
+
+    # keys
     KEY_REASON = "reason"
+    PLAN_UUID = "plan_uuid"
+    TIME_TAKEN_FOR_COMPILATION = "time_taken_for_compilation"
+    COMPLEXITY_SCORE_BOUNDS = "complexity_score_bounds"
+    BEFORE_COMPLEXITY_SCORE = "before_complexity_score"
```

Comment:
->complexity_score_before_compilation, and complexity_score_after_compilation to be more clear